### DISCUSSION TITLE
Using GitHub hosted NuGet Packages

### DISCUSSION THREAD

**Discussion by Brennan1994** [HydrologicEngineeringCenter/HEC-FDA#170] [Announcements]
To use the GitHub hosted NuGet packages. like fda-model, you'll need to authenticate. To do this, you need a personal access token. Instructions for creating that token are here: https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token

At a minimum, you'll need read access permissions. further Instructions for which permissions to assign that token available here:
https://docs.github.com/en/packages/learn-github-packages/about-permissions-for-github-packages

Once you're in Visual Studio, you'll want to add the HEC Github as a source. This can be done under the tools|options menu.
![image](https://user-images.githubusercontent.com/64556723/138994973-364aa1c0-c13d-494d-bb9e-6847906a4258.png)

Once added, You'll want to browse all the packages you now have access to. You can do this buy opening the package manager  from tools|manage packages for solution . Select the HEC Github from your dropdown menu in the top right, and click the browse tab. On your first time opening this you should get a prompt to enter a username and password. Enter the following:

Username: HydrologicEngineeringCenter
Password:{Your Personal Access Token}

When you finish. you should see all the available packages like this: 
![image](https://user-images.githubusercontent.com/64556723/138995384-c96a1c21-1610-43b7-9516-4554a75795d1.png)

**Comment by rnugent3** [comment]
I've referred back to this discussion so many times. @Brennan1994 thanks for documenting this for us.

**Comment by rnugent3** [comment]
When a token expires, sometimes VS will not prompt you to re-enter credentials. The way to fix this is to iteratively:

1. Close out of VS
2. Open Windows Credential Manager
3. Delete a related credential (e.g. related to NuGet) 
4. Open VS and navigate to manage NuGet packages for solution 
5. If credentials prompted, you're done. If not, repeat (1) through (4) for other related credentials.

**Reply by Brennan1994** [reply]
The offending credential is usually prefixed with VS (for visual studio) and says something like Github or NuGet in the name.
